sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcFileFormat.scala

```diff
@@ -118,6 +120,11 @@ class OrcFileFormat extends FileFormat with DataSourceRegister with Serializable
     val broadcastedHadoopConf =
       sparkSession.sparkContext.broadcast(new SerializableConfiguration(hadoopConf))
 
+    val enableVectorizedReader: Boolean =
+      sparkSession.sessionState.conf.orcVectorizedReaderEnabled &&
+      dataSchema.forall(f => f.dataType.isInstanceOf[AtomicType] &&
```

Comment:
This is not reliable. If a new `AtomicType` type gets introduced and not supported for vectorised reads, this will not guard us against that. Its easy for anyone to forget to add new atomic type to the exclusion list in this check